src/main/scala/is/hail/io/vcf/ExportVCF.scala

```diff
@@ -1,96 +1,106 @@
+        val x = m.loadInt(offset)
+        sb.append(x.toString)
+      case TInt64(_) =>
+        val x = m.loadLong(offset)
+        if (x > Int.MaxValue || x < Int.MinValue)
+          fatal(s"Cannot convert Long to Int if value is greater than Int.MaxValue (2^31 - 1) " +
+            s"or less than Int.MinValue (-2^31). Found $x.")
+        sb.append(x.toInt.toString)
+      case TFloat32(_) =>
+        val x = m.loadFloat(offset)
+        if (x.isNaN)
+          sb += '.'
+        else
+          sb.append(x.formatted("%.5e"))
+      case TFloat64(_) =>
+        val x = m.loadDouble(offset)
+        if (x.isNaN)
+          sb += '.'
+        else
+          sb.append(x.formatted("%.5e"))
+      case TString(_) =>
+        sb.append(TString.loadString(m, offset))
+      case TCall(_) =>
+        val p = Genotype.gtPair(m.loadInt(offset))
+        sb.append(s"${ p.j }/${ p.k }")
+      case _ =>
+        fatal(s"VCF does not support type $elementType")
     }
   }
-
-  def emitFormatField(f: Field, sb: StringBuilder, a: Annotation) {
-    f.typ match {
-      case TCall(_) => sb.append(Call.toString(a.asInstanceOf[Call]))
-      case it: TIterable =>
-        if (a == null)
+  
+  def iterableVCF(sb: StringBuilder, t: TIterable, m: MemoryBuffer, length: Int, offset: Long) {
+    if (length > 0) {
+      var i = 0
+      while (i < length - 1) {
+        if (t.isElementDefined(m, offset, i)) {
+          val eOffset = t.loadElement(m, offset, length, i)
+          strVCF(sb, t.elementType, m, eOffset)
+        } else
           sb += '.'
-        else {
-          val arr = a.asInstanceOf[Iterable[_]]
-          arr.foreachBetween(a => strVCF(sb, it.elementType, a))(sb += ',')
-        }
-      case t => strVCF(sb, t, a)
-    }
+        sb += ','
+        i += 1
+      }
+      if (t.isElementDefined(m, offset, i)) {
+        val eOffset = t.loadElement(m, offset, length, i)
+        strVCF(sb, t.elementType, m, eOffset)
+      }
+    } else
+      sb += '.'
   }
 
-  def emitInfo(f: Field, sb: StringBuilder, value: Annotation, wroteLast: Boolean): Boolean = {
-    if (value == null)
-      wroteLast
-    else
-      f.typ match {
-        case it: TIterable =>
-          val arr = value.asInstanceOf[Iterable[_]]
-          if (arr.isEmpty) {
-            wroteLast
-          } else {
-            if (wroteLast)
-              sb += ';'
-            sb.append(f.name)
-            sb += '='
-            arr.foreachBetween(a => strVCF(sb, it.elementType, a))(sb += ',')
-            true
-          }
-        case TBoolean(_) => value match {
-          case true =>
-            if (wroteLast)
-              sb += ';'
-            sb.append(f.name)
-            true
-          case _ =>
-            wroteLast
-        }
-        case t =>
+  def emitInfo(sb: StringBuilder, f: Field, m: MemoryBuffer, offset: Long, wroteLast: Boolean): Boolean = {
+    f.typ match {
+      case it: TIterable if it.elementType != TBoolean() =>
```

Comment:
if you don't want to filter for only optional booleans, I'd use `it.elementType.isOfType(TBoolean())`.